### PR TITLE
Replace SymbolLookupCache with comparing FullNames for symbols.

### PR DESCRIPTION
**Pull request by pranavkm**
This addresses an issue with ClassLIbraries containing tag helpers that target a different \ earlier version of Razor.Runtime than the one the application targets. The compilation graph ends up with two symbols (RetargatedTypeSymbol and PEBasedTypeSymbol) that represents the same type and consequently symbol comparisons fail. We'll instead simply compare FullNames for types and symbols.

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @NTaylorMullen

File: src/Microsoft.AspNet.Razor.Runtime.Precompilation/CodeAnalysisSymbolBasedTypeInfo.cs [hunk 1]
```diff
@@ -167,6 +163,23 @@ public ITypeInfo[] GetGenericDictionaryParameters()
         }
 
         /// <summary>
+        /// Gets a value that indicates if <paramref name="targetTypeInfo"/> is represented using
+        /// <paramref name="sourceTypeSymbol"/> in the symbol graph.
+        /// </summary>
+        /// <param name="sourceTypeSymbol">The <see cref="ITypeSymbol"/>.</param>
+        /// <param name="targetTypeInfo">The <see cref="System.Reflection.TypeInfo"/>.</param>
+        /// <returns></returns>
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNet.Razor.Runtime.Precompilation/CodeAnalysisSymbolBasedTypeInfo.cs]: Missing

File: src/Microsoft.AspNet.Razor.Runtime.Precompilation/CodeAnalysisAttributeUtilities.cs [hunk 2]
```diff
@@ -172,8 +165,9 @@ public static class CodeAnalysisAttributeUtilities
                     }
                     else
                     {
-                        var parameterSymbol = symbolLookup.GetSymbol(runtimeParameter.GetTypeInfo());
-                        if (symbolConstructorArguments[index].Type != parameterSymbol)
+                        if (!CodeAnalysisSymbolBasedTypeInfo.IsType(
+                                symbolConstructorArguments[index].Type,
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNet.Razor.Runtime.Precompilation/CodeAnalysisAttributeUtilities.cs]: nit: inconsistency with spacing here and on lines 159+160

File: src/Microsoft.AspNet.Razor.Runtime.Precompilation/CodeAnalysisSymbolBasedTypeInfo.cs [hunk 3]
```diff
@@ -24,7 +24,8 @@ public class CodeAnalysisSymbolBasedTypeInfo : ITypeInfo
         /// </summary>
         public static readonly System.Reflection.TypeInfo OpenGenericDictionaryTypeInfo =
             typeof(IDictionary<,>).GetTypeInfo();
-        private readonly CodeAnalysisSymbolLookupCache _symbolLookup;
+        private static readonly System.Reflection.TypeInfo TagHelperTypeInfo =
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNet.Razor.Runtime.Precompilation/CodeAnalysisSymbolBasedTypeInfo.cs]: Can't just do `TypeInfo`?

**Comment by NTaylorMullen** [comment]
:shipit:

File: src/Microsoft.AspNet.Razor.Runtime.Precompilation/CodeAnalysisSymbolBasedTypeInfo.cs (hunk 3 above)
Comment (pranavkm) [inline src/Microsoft.AspNet.Razor.Runtime.Precompilation/CodeAnalysisSymbolBasedTypeInfo.cs]: Unfortunately not. There's a TypeInfo in Roslyn.

File: src/Microsoft.AspNet.Razor.Runtime.Precompilation/CodeAnalysisSymbolBasedTypeInfo.cs (hunk 1 above)
Comment (pranavkm) [inline src/Microsoft.AspNet.Razor.Runtime.Precompilation/CodeAnalysisSymbolBasedTypeInfo.cs]: :+1:

**Pull request closed without merging**